### PR TITLE
EVG-7465 Refuse to enqueue an empty patch

### PR DESCRIPTION
**Pull request by malikchaya2**
[EVG-7465](https://jira.mongodb.org/browse/EVG-7465)

### CODE REVIEW

File: operations/commit_queue_test.go [hunk 1]
```diff
@@ -142,6 +143,37 @@ func (s *CommitQueueSuite) TestListContentsForCLI() {
...
+		Project:     "mci",
+		Author:      "annie.black",
+		Activated:   true,
+		Description: "fix things",
+		CreateTime:  now,
+		Status:      evergreen.TaskDispatched,
+	}
+	s.NoError(p1.Insert())
+
+	pRef := &model.ProjectRef{
```
Comment (ybrill) [inline operations/commit_queue_test.go]: do we need a projectRef for this test?

File: operations/commit_queue_test.go [hunk 2]
```diff
@@ -142,6 +143,37 @@ func (s *CommitQueueSuite) TestListContentsForCLI() {
 	s.Contains(stringOut, patchURL)
 }
 
+func (s *CommitQueueSuite) TestMergeWithEmptyPatch() {
+	s.Require().NoError(db.ClearCollections(commitqueue.Collection, patch.Collection, model.ProjectRefCollection))
```
Comment (ybrill) [inline operations/commit_queue_test.go]: do we need to clear the `commitqueue` collection?

File: operations/commit_queue_test.go (hunk 1 above)
Comment (malikchaya2) [reply operations/commit_queue_test.go]: I need the mci project in order to create the patch

File: operations/commit_queue_test.go (hunk 2 above)
Comment (malikchaya2) [reply operations/commit_queue_test.go]: It is not! thanks you!

File: operations/commit_queue_test.go [hunk 3]
```diff
@@ -142,6 +143,37 @@ func (s *CommitQueueSuite) TestListContentsForCLI() {
...
+	s.Require().NoError(pRef.Insert())
+
+	mergeParam := mergeParams{
+		projectID: "some id",
+		pause:     false,
+		id:        "aabbccddeeff112233445566",
+	}
+	ac, _, err := s.conf.getLegacyClients()
+	s.NoError(err)
+	err = mergeParam.mergeBranch(s.ctx, s.conf, s.client, ac)
```
[the remark this answers is not in the document]
Comment (malikchaya2) [reply operations/commit_queue_test.go]: It panics if ac is nil

File: operations/commit_queue.go [hunk 4]
```diff
@@ -302,6 +302,13 @@ func (p *mergeParams) mergeBranch(ctx context.Context, conf *ClientSettings, cli
 	if p.pause {
 		return nil
 	}
+	patchDoc, err := ac.GetPatch(p.id)
+	if err != nil {
+		return err
+	}
+	if len(patchDoc.Patches) == 0 {
```
[the remark this answers is not in the document]
Comment (malikchaya2) [reply operations/commit_queue.go]: It seems to be cast to a patch.Patch over [here]( https://github.com/evergreen-ci/evergreen/blob/55aabe746373abc9a89a1be50ce40077640ce179/operations/http.go#L213). Does that makes sense?

[the remark this answers is not in the document]
Comment (ybrill) [reply operations/commit_queue.go]: Although it does end up in a patch.Patch the data in the patch.Patch is filled in by the ToService which doesn't copy over the `Patches`

File: operations/commit_queue_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (ybrill) [reply operations/commit_queue_test.go]: why?

File: operations/commit_queue.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (malikchaya2) [reply operations/commit_queue.go]: Ah! thanks for pointing that out!

File: operations/commit_queue_test.go (hunk 1 above)
Comment (ybrill) [reply operations/commit_queue_test.go]: Is there an execution path where we check the project of the patch? Could it be omitted?

File: operations/commit_queue_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (malikchaya2) [reply operations/commit_queue_test.go]: I'm not sure, but when I set it to nil, I get:
mongo: no documents in result
and 
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0x5cceb63]

[the remark this answers is not in the document]
Comment (ybrill) [reply operations/commit_queue_test.go]: I see now. My mistake (I was looking at my code, not the code in the PR 😳)

File: operations/commit_queue_test.go (hunk 1 above)
Comment (ybrill) [reply operations/commit_queue_test.go]: resolved in person.

**Pull request closed without merging**